packages/react-utilities/src/compose/types.ts

```diff
@@ -29,17 +29,27 @@ export type DefaultObjectShorthandProps = ObjectShorthandProps<
 >;
 
 /**
- * Defines the slot props for a slot that supports a Component type.
- *
- * For intrinsic elements like 'div', use {@link IntrinsicShorthandProps} instead.
+ * Takes the props we want to support for a slot and adds the ability for `children` to be a render function that takes
+ * those props.
  */
 export type ObjectShorthandProps<Props extends { children?: React.ReactNode } = {}> = Props & {
   children?: Props['children'] | ShorthandRenderFunction<Props>;
 };
 
+/**
+ * Defines the slot props for a slot that supports a Component type.
+ *
+ * For intrinsic elements like 'div', use {@link IntrinsicShorthandProps} instead.
+ */
```

Comment:
Maybe call out that the template argument needs `typeof`, and include a usage example since it's not straightforward. E.g.:
```suggestion
/**
 * Defines the slot props for a slot that supports a Component type.
 *
 * For intrinsic elements like 'div', use {@link IntrinsicShorthandProps} instead.
 *
 * The template param is the type of a control, using the `typeof` keyword. For example:
 * ```
 * ComponentSlotProps<typeof Button>
 * ```
 */
```

Reply:
This is great, I'll change this.